ocs_ci/ocs/cluster.py

```diff
@@ -878,3 +878,135 @@ def validate_pg_balancer():
             return False
     else:
         logging.info(f"pg_balancer is not active")
+
+
+def get_child_nodes_osd_tree(node_id, osd_tree):
+    """
+    This functions finds the children of a node from the 'ceph osd tree' and returns them as list
+
+    Args:
+        node_id (int): the id of the node for which the children to be retrieved
+        osd_tree (dict): dictionary containing the output of 'ceph osd tree'
+
+    Returns:
+        list: of 'children' of a given node_id
+    """
+    for i in range(len(osd_tree['nodes'])):
+        if osd_tree['nodes'][i]['id'] == node_id:
+            return osd_tree['nodes'][i]['children']
+
+
+def check_osds_in_hosts_osd_tree(all_hosts, osd_tree):
+    for each_host in range(len(all_hosts)):
+        osd_in_each_host = get_child_nodes_osd_tree(all_hosts[each_host], osd_tree)
+        print("osd = ", osd_in_each_host)
+        if len(osd_in_each_host) > 1 or len(osd_in_each_host) <= 0:
+            logging.error("Error. ceph osd tree is NOT formed correctly after cluster expansion")
+            return False
+
+    logging.info(f"osd tree verification Passed")
+    return True
+
+
+def check_osd_tree_1az_vmware(osd_tree, number_of_osds):
+    """
+    Checks whether an OSD tree is created/modified correctly. This can be used as a verification step for
+    deployment and cluster expansion tests.
+    This function is specifically for ocs cluster created on 1 AZ VMWare setup
+
+    Args:
+        osd_tree(dict): Dictionary of the values which represent 'osd tree'.
+        number_of_osds(int): total number of osds in the cluster
+
+    Returns:
+        Boolean: True, if the ceph osd tree is formed correctly. Else False
+    """
+    # in case of vmware, there will be only one zone as of now. The OSDs are arranged as follows:
+    # ID  CLASS WEIGHT  TYPE NAME                            STATUS REWEIGHT PRI-AFF
+    # -1       0.99326 root default
+    # -8       0.33109     rack rack0
+    # -7       0.33109         host ocs-deviceset-0-0-dktqc
+    #  1   hdd 0.33109             osd.1                        up  1.00000 1.00000
+    # There will be 3 racks - rack0, rack1, rack2.
+    # When cluster expansion is successfully done, a host and an osd are added in each rack.
+    # The number of hosts will be equal to the number osds the cluster has. Each rack can
+    # have multiple hosts but each host will have only one osd under it.
+    number_of_hosts_expected = number_of_osds/3
+    all_hosts = []
+    racks = osd_tree['nodes'][0]['children']
+
+    for rack in range(len(racks)):
```

Comment:
It will be more clear to change the for loop to this: 'for rack in racks:' and then in the code below to replace 'racks[rack]' with 'rack'. Same issue in the functions: 'check_osd_tree_3az_aws', 'check_osd_tree_1az_aws'.